sql/sql_statistics.cc

```diff
@@ -1086,7 +1086,7 @@ class Column_stat: public Stat_table
           // Note: this is dumb. the histogram size is stored with the
           // histogram!
           stat_field->store(stats->histogram_? 
-                              stats->histogram_->get_size() : 0);
+                              stats->histogram_->get_width() : 0);
           break;
```

Comment:
Do I read it correctly: this function now stores the number of buckets, not the number of bytes?
Please don't do this as this breaks compatibility with older code.

Instead, please check the length of the data that was written by `stats->histogram_->serialize(stat_field);` and store that (yes, this means you need to process COLUMN_STAT_HISTOGRAM column before the COLUMN_STAT_HIST_SIZE. Please change the code in this function to do that)

Reply:
I think it is also fine to make `histogram_->serialize` return the number of bytes stored.